lib/vagrant-ovirt4/action/create_network_interfaces.rb

```diff
@@ -27,9 +27,19 @@ def call(env)
             iface_options = scoped_hash_override(options, :ovirt)
           end
 
-          raise Errors::NoNetworkError if iface_options[:network_name].nil? 
+          raise Errors::NoNetworkError if iface_options[:network_name].nil?
+          config = env[:machine].provider_config
+          @logger.info("Finding network #{iface_options[:network_name]} for given cluster #{config.cluster}")
+          network_profile_id = nil
+          clusters_service = env[:connection].system_service.clusters_service
+          cluster = clusters_service.list(search: "name=#{config.cluster}").first
           profiles_service = env[:connection].system_service.vnic_profiles_service
-          network_profile_id = profiles_service.list.map { |p| p.id if p.name == iface_options[:network_name] }.first
+          profiles_service.list(search: "name=#{iface_options[:network_name]}").each do |profile|
```

Comment:
I leave  this up to you. I can merge as-is or you can rewrite this block to 

`network_profile_id = profiles_service.list(search: "name=#{iface_options[:network_name]}").map { |profile| profile.id if env[:connection].follow_link(profile.network).data_center.id == cluster.data_center.id }.compact.first`

It's a little less readable so I leave that choice to you